utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -12,60 +12,115 @@
 import platform
 
 
-class StdlibDeploymentTarget(object):
+class Platform(object):
+    """
+    Abstract representation of a platform Swift can run on.
+    """
+
+    def __init__(self, name, archs, sdk_name=None, is_darwin=False):
+        """
+        Create a platform with the given name and list of architectures.
+        """
+        self.name = name
+        self.targets = [Target(self, arch) for arch in archs]
```

Comment:
I'm with @ddunbar on this one.  We can have multiple variants of a single architecture; especially for arm we need to have provisions to support the hard-float ABI, or we will end up with hacks later.